DietPi-Sofware | MediaWiki: A collaboration and documentation platform

Joulinar opened the pull request:
- add new software title ID 141

Joulinar committed:
v8.19

DietPi-Sofware | MediaWiki - add new software title

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - shellcheck

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - shellcheck

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -869,6 +869,13 @@ Available commands:
+		#------------------
+		software_id=141
+		aSOFTWARE_NAME[$software_id]='MediaWiki'
+		aSOFTWARE_DESC[$software_id]='A collaboration and documentation platform'
```
Shall we add the info that Wikipedia is using it?

Joulinar replied:
at least on the docs we can do 😄

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - For MediaWiki, at least PHP 7.4.3 is required. Therefore, installation on Buster is not supported.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11652,6 +11659,43 @@ _EOF_
+			local version="1.39"
+			local file="mediawiki-1.39.3"
+			local fallback_url="https://releases.wikimedia.org/mediawiki/1.39/mediawiki-1.39.3.tar.gz"
+			Download_Install "https://releases.wikimedia.org/mediawiki/$version/$file.tar.gz"
```
```suggestion
			local version='1.39.3'
			Download_Install "https://releases.wikimedia.org/mediawiki/${version%.*}/mediawiki-$version.tar.gz"
```
- Probably we can at least get the latest patch version from the downloads, so we only need to regularly check for new minor versions: https://releases.wikimedia.org/mediawiki/1.39/
- Not sure whether it is safe to assume the second latest minor version to be the latest stable one? https://releases.wikimedia.org/mediawiki/
- We can check once v1.40 has been released whether there is a `/1.41/` directory immediately.
- Last but not least we could scrape the latest version from the main download page: https://www.mediawiki.org/wiki/Download

Joulinar replied:
Yes, I used the hardcode values for testing. I was hoping for some clever ideas from your side on how best to retrieve the correct version. Good idea to accept the second newest minor version as the newest stable version.

MichaIng replied:
I will have a look about whether it can be well obtained from the main download page. Otherwise I think we should wait for v1.40 release to see whether a v1.41 directory really already exists at this point.

I mean we could also check the newest first, see whether there is a download listed without `-rc` or any such suffix, else check next dir. But IMO too complicated, requiring too many network requests etc.

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11652,6 +11659,43 @@ _EOF_
+			Download_Install "https://releases.wikimedia.org/mediawiki/$version/$file.tar.gz"
+
+			# Reinstall: Clean install but preserve existing config file
+			[[ -f '/var/www/wiki/LocalSettings.php' ]] && G_EXEC mv /var/www/wiki/LocalSettings.php "$file/"
```
```suggestion
			[[ -f '/var/www/wiki/LocalSettings.php' ]] && G_EXEC mv /var/www/wiki/LocalSettings.php "mediawiki-$version/"
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11652,6 +11659,43 @@ _EOF_
+
+			# Move new instance in place
+			G_EXEC cp -a "$file/." /var/www/wiki/
+			G_EXEC rm -R "$file"
```
```suggestion
			G_EXEC cp -a "mediawiki-$version/." /var/www/wiki/
			G_EXEC rm -R "mediawiki-$version"
```

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11652,6 +11659,43 @@ _EOF_
+			G_EXEC cp -a "$file/." /var/www/wiki/
+			G_EXEC rm -R "$file"
+
+			unset -v version file
```
```suggestion
			unset -v version
```
Okay, as `$file` was used quite often, probably not a bad idea to define it as well, but based on `$version` so that only a single variable needs to be updated manually.

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - small adjustemt

Co-authored-by: MichaIng <micha@dietpi.com>

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - small adjustment

Co-authored-by: MichaIng <micha@dietpi.com>

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - small adjustment

Co-authored-by: MichaIng <micha@dietpi.com>

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - small adjustment on the way how latest version is checked.

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng commented on dietpi/dietpi-software:
```diff
@@ -11652,6 +11661,41 @@ _EOF_
+			G_EXEC cp -a "mediawiki-$version/." /var/www/wiki/
+			G_EXEC rm -R "mediawiki-$version"
+
+			unset -v version
```
I think using the main download page is quite easiest here:
```suggestion
			local fallback_url='https://releases.wikimedia.org/mediawiki/1.39/mediawiki-1.39.3.tar.gz'
			Download_Install "$(curl -sSfL 'https://www.mediawiki.org/wiki/Download' | grep -o 'https://releases\.wikimedia\.org/mediawiki/[^/"]*/mediawiki-[^"]*\.tar\.gz' | head -1)"

			# Reinstall: Clean install but preserve existing config file
			[[ -f '/var/www/wiki/LocalSettings.php' ]] && G_EXEC mv /var/www/wiki/LocalSettings.php mediawiki-*/
			[[ -d '/var/www/wiki' ]] && G_EXEC rm -R /var/www/wiki

			# Move new instance in place
			G_EXEC mv mediawiki-* /var/www/wiki
```

Joulinar committed:
v8.19

DietPi-Software | MediaWiki - adjust the way fetching current version.

Co-authored-by: MichaIng <micha@dietpi.com>

MichaIng committed:
v8.19

- CI | DietPi-Software: Test webserver, PHP and database when installing MediaWiki

MichaIng committed:
v8.19

- DietPi-Software | MediaWiki: Do all file writes in tmpfs before moving install to disk, switch to software ID 210 to not remove and readd a software title with the same ID with some release, which makes install state cleanup difficult

MichaIng committed:
v8.19

- DietPi-Software/DietPi-LetsEncrypt | Use shipped Lighttpd setenv module config from Bullseye on where required
- DietPi-Software | MediaWiki: Set nosniff header for uploads to mute warning on Lighttpd and Nginx, set OOTB on Apache
